typing/typecore.ml

```diff
@@ -986,6 +987,8 @@ let solve_Ppat_construct ~refine tps penv loc constr no_existentials
 
 let solve_Ppat_record_field ~refine loc penv label label_lid record_ty =
   with_local_level_generalize_structure begin fun () ->
+    if not refine && label.lbl_atomic = Atomic then
```

Comment:
The `not refine` test tells us that we are type-checking a field pattern coming from user code, rather than checking a counter-example candidate for exhaustivity checking. I would propose a comment such as follows to clarify this:

```ocaml
(* [not refine] holds when the pattern comes from the user,
  rather than a counter-example to exhaustivity. *)